compiler/rustc_borrowck/src/lib.rs

```diff
@@ -584,6 +566,49 @@ struct MirBorrowckCtxt<'a, 'infcx, 'tcx> {
     move_errors: Vec<MoveError<'tcx>>,
 }
 
+impl<'a, 'infcx, 'tcx> MirBorrowckCtxt<'a, 'infcx, 'tcx> {
+    fn new(
```

Comment:
I don't think this is cleaner, tbh. I think initializing `MirBorrowckCtxt` with the struct initialization context was totally fine, and now we're just roundtripping through a `new` method that needs to stay in sync with the billion fields that the struct has.

Reply:
Ok, I'll remove it.